File: apple/Sources/ReactNativeWatchOS/Modifiers/OnDisappear.swift
import SwiftUI

/// SwiftUI `.onDisappear(perform:)`. Fires the JS `handler` when the view
/// disappears.
enum RNWOnDisappearModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("onDisappear") { view, params, ctx in
            AnyView(view.onDisappear { ctx.fire(params.int("handler")) })
        }
    }
}

File: src/modifiers/onDisappear.ts
import { createModifier } from './createModifier';

/// SwiftUI `.onDisappear(perform:)`. Fires `handler` when the view disappears.
///
/// ```tsx
/// <Text modifiers={[onDisappear(() => stopTimer())]} />
/// ```
export function onDisappear(handler: () => void) {
  return createModifier('onDisappear', { handler });
}
